src/ot_orchestration/dags/platform_input_support.py

```diff
@@ -1,96 +1,102 @@
 
-from airflow.decorators import task
+from airflow.decorators.task_group import task_group
 from airflow.models.dag import DAG
-from airflow.models.taskinstance import TaskInstance
-from airflow.providers.google.cloud.operators.cloud_run import (
-    CloudRunCreateJobOperator,
-    CloudRunDeleteJobOperator,
+from airflow.models.param import Param
+from airflow.operators.empty import EmptyOperator
+from airflow.providers.google.cloud.operators.compute import (
+    ComputeEngineDeleteInstanceOperator,
 )
-from airflow.utils.task_group import TaskGroup
+from airflow.utils.edgemodifier import Label
 
-from ot_orchestration.operators.cloud_run_fetch_logs_operator import (
-    CloudRunExecuteJobWithLogsOperator,
-)
-from ot_orchestration.utils.cloud_run import clean_name, create_cloud_run_job, strhash
+from ot_orchestration.operators.gce import ComputeEngineRunContainerizedWorkloadSensor
+from ot_orchestration.operators.gcs import UploadFileOperator
+from ot_orchestration.operators.platform import PISDiffComputeOperator
+from ot_orchestration.utils import clean_name, read_yaml_config
 from ot_orchestration.utils.common import (
+    GCP_PROJECT_PLATFORM,
     GCP_REGION,
     platform_dag_kwargs,
     shared_dag_args,
 )
-from ot_orchestration.utils.utils import read_yaml_config
-
-PIS_CONFIG_PATH = Path(__file__).parent / "config" / "pis.yaml"
-PIS_IMAGE = "europe-west1-docker.pkg.dev/open-targets-eu-dev/platform-input-support-test/platform-input-support-test:latest"
-PIS_GCP_PROJECT = "open-targets-eu-dev"
-PIS_MACHINE_SPEC = {"cpu": "1", "memory": "512Mi"}
-
-
-def get_steps() -> list[str]:
-    """Read the steps from the PIS configuration file."""
-    yaml_config = read_yaml_config(PIS_CONFIG_PATH)
-    return yaml_config["steps"].keys()
 
+PIS_IMAGE = "europe-west1-docker.pkg.dev/open-targets-eu-dev/platform-input-support-test/platform-input-support-test"
+PIS_SACC = "platform-input-support@open-targets-eu-dev.iam.gserviceaccount.com"
+CONFIG_PATH = Path(__file__).parent / "config" / "pis.yaml"
+CONFIG = read_yaml_config(CONFIG_PATH)
+GCS_URL = CONFIG["gcs_url"]
 
 with DAG(
-    "platform_input_support",
+    dag_id="platform_input_support",
     default_args=shared_dag_args,
     description="Open Targets Platform — platform-input-support",
     **platform_dag_kwargs,
+    params={
+        "run": Param(
+            f"pis-{datetime.now().strftime('%Y%m%d-%H%M')}",
+            description="Unique run identifier, will be added as a label to infrastructure resources.",
+        ),
+        "pis_image_tag": Param(
+            "latest",
+            description=(
+                "The tag of the platform-input-support image to use. Defaults to 'latest', "
+                "but it is recommended to use a specific tag. Keep in mind tags do not have the 'v' prefix."
+            ),
+        ),
+    },
 ) as dag:
-    for step_name in get_steps():
-        clean_step_name = clean_name(step_name)
+    for step in list(CONFIG["steps"].keys()):
```

Comment:
I thought about it, but the config can change between executions (not runs, but partial reruns of some failed steps). If I put the config in a param it will be immutable for the whole run, and then you'd have to rerun the whole DAG if something requiring a config change fails.

Reply:
You can specify it when triggering the dag then. The values you pass to the params are only the defaults in that context.

Reply:
Imagine this situation:

You run the Pipeline DAG with a config for PIS which fails for one step only. That step is left red in the DAG and you want to rerun it after editing the config file.

If you pass the config into the DAG as a param, it will be the same when we go into Airflow UI and clean the failed step so it runs again. The changes will not be picked up and it will fail again.

Reply:
You almost convinced me. Let's keep it outside, although I want to test this behavior.

Reply:
You can see the values for the params for a given run in the 'details' tab when clicking on a run. Those will be fixed for the whole run and will not change if you rerun parts of the graph.

![image](https://github.com/user-attachments/assets/dced7abb-0ac3-4a79-8da5-e7e69ab228aa)